### PR TITLE
Add an option to disable Service creation and define a name to use

### PR DESCRIPTION
**Pull request by galibey**
This might be useful for heterogeneous clusters. For example, you wish
to install two independent statefulsets with different
configurations but Hazelcast members are supposed to form one cluster.
To do this, you can install two helm releases, but in the second one
the Service creation turned off and used the service name from the first
release. Hazelcast members will be able to find each other by Hazelcast
Service Discovery Kubernetes plugin.

### CODE REVIEW

File: stable/hazelcast/templates/_helpers.tpl [hunk 1]
```diff
@@ -42,6 +42,17 @@ Create the name of the service account to use
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create the name of the service to use
+*/}}
+{{- define "hazelcast.serviceName" -}}
+{{- if .Values.service.create -}}
+    {{ default (include "hazelcast.fullname" .) .Values.service.name }}
```
Comment (leszko) [inline stable/hazelcast/templates/_helpers.tpl]: Why do you concatenate with `.Values.service.name` here?

I think the use cases are that you either create a service (then, it's just `"hazelcast.fullname"`) or you use the specified `.Values.service.name`.

File: stable/hazelcast/values.yaml [hunk 2]
```diff
@@ -103,6 +103,12 @@ readinessProbe:
 
 # Hazelcast Service properties
 service:
+  # Specifies whether a Service should be created
+  create: true
+  # The name of the Service to use.
+  # The name is used for service discovery by Hazelcast K8S discovery plugin.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
```
Comment (leszko) [inline stable/hazelcast/values.yaml]: How about commenting out `name` as we do with other non-used-by-default parameters?

File: stable/hazelcast/templates/_helpers.tpl (hunk 1 above)
Comment (galibey) [reply stable/hazelcast/templates/_helpers.tpl]: I thought that an option to create the service with specified name would be also useful. But, may be not though. I removed it.

File: stable/hazelcast/values.yaml (hunk 2 above)
Comment (galibey) [reply stable/hazelcast/values.yaml]: Fixed

**Comment by galibey** [comment]
Added the same changes to hazelcast-enterprise chart and bumped up versions.

**Comment by MayurDhande-GSLab** [comment]
Hello @galibey, it seems you made this change for the heterogenous clusters. But what about having one single helm chart of hazelcast, in which I want name of the service to be different than what is autogenerated (without having to rename the release) ?
Basically I want to use `service.name` to set the name of k8 service object.

**Comment by MayurDhande-GSLab** [comment]
And by the way, enterprise documentation does not reflect this change,

image

**Pull request merged**